Add vmap support for slogdet; fix regression from functorch 0.2.1

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #86823
* __->__ #86815

This PR adds vmap support for slogdet -- slogdet just decomposes into
linalg.slogdet.

This fixes a regression from functorch 0.2.1 (slogdet had a batching
rule then, and doesn't anymore). We didn't catch the regression because
it seems like slogdet doesn't have an OpInfo (I'm not sure if it had one
before).

Test Plan:
- new one-off test.

zou3519 committed:
Add vmap support for slogdet; fix regression from functorch 0.2.1

This PR adds vmap support for slogdet -- slogdet just decomposes into
linalg.slogdet.

This fixes a regression from functorch 0.2.1 (slogdet had a batching
rule then, and doesn't anymore). We didn't catch the regression because
it seems like slogdet doesn't have an OpInfo (I'm not sure if it had one
before).

Test Plan:
- new one-off test.

[ghstack-poisoned]

samdow commented on test/functorch/test_vmap.py:
```diff
@@ -3538,6 +3538,12 @@ def compute_A(out):
+        # There's no OpInfo for this
+        B = 2
+        x = torch.randn(2, 5, 5, device=device)
+        self.vmap_outplace_test(torch.slogdet, (x,), {}, (0,))
```
Do we also want the `check_vmap_fallback` since IIUC it was working before but raising the warning

zou3519 replied:
Good catch. I forgot that vmap_outplace_test doesn't do the check_vmap_fallback

zou3519 committed:
Update on "Add vmap support for slogdet; fix regression from functorch 0.2.1"


This PR adds vmap support for slogdet -- slogdet just decomposes into
linalg.slogdet.

This fixes a regression from functorch 0.2.1 (slogdet had a batching
rule then, and doesn't anymore). We didn't catch the regression because
it seems like slogdet doesn't have an OpInfo (I'm not sure if it had one
before).

Test Plan:
- new one-off test.

[ghstack-poisoned]

zou3519 commented:
@pytorchbot merge -f "unrelated failure"
